localstack-core/localstack/services/apigateway/next_gen/execute_api/parameters_mapping.py

```diff
@@ -0,0 +1,160 @@
+# integration response data, including the other data, to the method response parameters. The method request
+# data includes request parameters (path, query string, headers) and the body. The integration response data
+# includes response parameters (headers) and the body. For more information about using the stage variables,
+# see API Gateway stage variables reference.
+#
+# https://docs.aws.amazon.com/apigateway/latest/developerguide/request-response-data-mappings.html
+import json
+from typing import TypedDict
+
+from localstack.utils.json import extract_jsonpath
+from localstack.utils.strings import to_str
+
+from .context import InvocationRequest
+from .gateway_response import Default4xxError
+from .variables import ContextVariables
+
+
+class RequestDataMapping(TypedDict):
+    # Integration request parameters, in the form of path variables, query strings or headers, can be mapped from any
+    # defined method request parameters and the payload.
+    header: dict[str, str]
+    path: dict[str, str]
+    querystring: dict[str, str | list[str]]
+
+
+class ParametersMapper:
+    def map_integration_request(
+        self,
+        request_parameters: dict[str, str],
+        invocation_request: InvocationRequest,
+        context_variables: ContextVariables,
+        stage_variables: dict[str, str],
+    ) -> RequestDataMapping:
+        request_data_mapping = RequestDataMapping(
+            header={},
+            path={},
+            querystring={},
+        )
+        # TODO: maybe extract functionality and re-use in `map_integration_response`
+
+        for integration_mapping, request_mapping in request_parameters.items():
+            integration_param_location, param_name = integration_mapping.removeprefix(
+                "integration.request."
+            ).split(".")
+
+            if request_mapping.startswith("method.request."):
+                method_req_expr = request_mapping.removeprefix("method.request.")
+                value = self._retrieve_parameter_from_invocation_request(
+                    method_req_expr, invocation_request
+                )
+
+            elif request_mapping.startswith("context."):
+                context_var_expr = request_mapping.removeprefix("context.")
+                value = self._retrieve_parameter_from_context_variables(
+                    context_var_expr, context_variables
+                )
+
+            elif request_mapping.startswith("stageVariables."):
+                stage_var_name = request_mapping.removeprefix("stageVariables.")
+                value = self._retrieve_parameter_from_stage_variables(
+                    stage_var_name, stage_variables
+                )
+
+            elif request_mapping.startswith("'") and request_mapping.endswith("'"):
+                value = request_mapping.strip("'")
+
+            else:
+                # TODO: verify this
+                value = None
```

Comment:
Should we raise or log? Could help some strange debugging in the future?

Reply:
We always could, but this should be caught by the provider CRUD level. Also, setting the value to `None` makes the value not go anywhere. I can add a logging statement I think!